server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanCommunicationDetailsCacheTest.java

```diff
@@ -61,4 +61,54 @@ public void testSingleConsumerFound() {
         assertEquals(cd, cdc.get(null, "id1"));
     }
 
+    @Test
+    public void testGetByIdMultiple() throws CacheException {
+        InfinispanCommunicationDetailsCache cdc = createCache();
+
+        CommunicationDetails cdMultipleConsumer1 = new CommunicationDetails();
+        cdMultipleConsumer1.setId("id1");
+        cdMultipleConsumer1.setTargetFragmentId("fragmentId1");
+        cdMultipleConsumer1.setMultiConsumer(true);
+
+        CommunicationDetails cdMultipleConsumer2 = new CommunicationDetails();
+        cdMultipleConsumer2.setId("id1");
+        cdMultipleConsumer2.setTargetFragmentId("fragmentId2");
+        cdMultipleConsumer2.setMultiConsumer(true);
+
+        CommunicationDetails cd3 = new CommunicationDetails();
+        cd3.setId("id2");
+
+        cdc.store(null, Arrays.asList(cdMultipleConsumer1, cdMultipleConsumer2, cd3));
+
+        List<CommunicationDetails> cdFromCache = cdc.getById(null, "id1");
+
+        Assert.assertEquals(2, cdFromCache.size());
+        Assert.assertEquals(new HashSet<>(Arrays.asList(cdMultipleConsumer1, cdMultipleConsumer2)),
+                new HashSet<>(cdFromCache));
+    }
+
+    @Test
+    public void testGetMultipleEmpty() {
+        InfinispanCommunicationDetailsCache cdc = createCache();
+
+        List<CommunicationDetails> id = cdc.getById(null, "id");
+        Assert.assertTrue(id.isEmpty());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testGetMultipleNull() {
+        InfinispanCommunicationDetailsCache cdc = createCache();
+
+        List<CommunicationDetails> id = cdc.getById(null, null);
+        Assert.assertTrue(id.isEmpty());
+    }
+
+
+    private InfinispanCommunicationDetailsCache createCache() {
+        InfinispanCommunicationDetailsCache cdc = new InfinispanCommunicationDetailsCache();
+        cdc.setCommunicationDetails(cacheManager.getCache(InfinispanCommunicationDetailsCache.CACHE_NAME +
+                UUID.randomUUID()));
+
```

Comment:
As per your discussion while I was away, I think changing to provide the cache in the constructor is better - as long as the default constructor exists, to enable the InfinispanCommsDetailsCache to be created by the ServiceLoader/cdi.